client/source/js/modules/analysis/optimization-ctrl.js

```diff
@@ -275,10 +277,129 @@ define(['./module', 'angular', 'd3'], function (module, angular, d3) {
       }
     };
 
+    function validateYears(){
+      return $scope.params.objectives!==undefined &&
+             $scope.params.objectives.year!==undefined &&
+             $scope.params.objectives.year.start!==undefined &&
+             $scope.params.objectives.year.end !==undefined &&
+             ($scope.params.objectives.year.start < $scope.params.objectives.year.end);
+    }
+
+    /**
+     * If the string is undefined return empty, otherwise just return the string
+     * @param str
+     * @returns {string}
+     */
+    function strOrEmpty(str){
+      return _(str).isUndefined() ? '' : str;
```

Comment:
I've used 
something || '' 
in other projects before and believe me that can cause more conflicts and headaches than explicit checking if something is undefined or not (also, the underscore function checks if the object equals "void 0", which is a better way to check for undefined.